sc2ts/info.py

```diff
@@ -1751,38 +1756,43 @@ def draw_subtree(
                     shown_nodes.append(u)
                 prev = u
 
-        if mutation_labels is None:
-            mutation_labels = collections.defaultdict(list)
-            multiple_mutations = []
-            reverted_mutations = []
-            use_mutations = np.where(np.isin(ts.mutations_node, shown_nodes))[0]
-            sites = ts.mutations_site[use_mutations]
-            for mut_id in use_mutations:
-                # TODO Viz the recurrent mutations
-                mut = ts.mutation(mut_id)
-                site = ts.site(mut.site)
-                if np.sum(sites == site.id) > 1:
-                    multiple_mutations.append(mut.id)
-                inherited_state = site.ancestral_state
-                if mut.parent >= 0:
-                    parent = ts.mutation(mut.parent)
-                    inherited_state = parent.derived_state
-                    parent_inherited_state = site.ancestral_state
-                    if parent.parent >= 0:
-                        parent_inherited_state = ts.mutation(
-                            parent.parent
-                        ).derived_state
-                    if parent_inherited_state == mut.derived_state:
-                        reverted_mutations.append(mut.id)
-                # Reverse map label name to mutation id, so we can count duplicates
-                label = f"{inherited_state}{int(site.position)}{mut.derived_state}"
-                mutation_labels[label].append(mut.id)
+        multiple_mutations = []
+        reverted_mutations = []
+        recurrent_mutations = collections.defaultdict(list)
+        mut_labels = {} if mutation_labels is None else mutation_labels.copy()
+        use_mutations = np.where(np.isin(ts.mutations_node, shown_nodes))[0]
+        sites = ts.mutations_site[use_mutations]
+        for mut_id in use_mutations:
```

Comment:
I don't think you need to do a lot of this work here - the TreeInfo class has the ``mutations_inherited_state`` and ``mutations_derived_state`` arrays defined, as well as ``mutation_is_reversion``, etc. Sorry I didn't spot this earlier.

Reply:
Ah neat, I didn't realise. Thanks. We should really get https://github.com/tskit-dev/tskit/issues/2631 into the next tskit version, but I've been putting it off because of https://github.com/tskit-dev/tskit/issues/2923#issuecomment-2049166759